services/traction/bdd-tests/features/steps/security.py

```diff
@@ -0,0 +1,26 @@
+import requests
+import json
+
+from behave import *
+from starlette import status
+
+
+@then(
+    '"{tenant_1}" cannot NOT get "{tenant_2}"s contact to "{tenant_2_contact_alias}" by id'
+)
+def step_impl(context, tenant_1: str, tenant_2: str, tenant_2_contact_alias: str):
+
+    response1 = requests.get(
+        context.config.userdata.get("traction_host") + "/tenant/v1/contacts",
+        params={"alias": tenant_2_contact_alias},
+        headers=context.config.userdata[tenant_2]["auth_headers"],
+    )
+    assert response1.status_code == status.HTTP_200_OK, response1.__dict__
+    tenant_2_contact = json.loads(response1.content)["items"][0]
+
+    response2 = requests.get(
```

Comment:
i agree that both list/search and get need to be tested, but probably better if list (tenant 1) and list (tenant 2) are compared and get (tenant 1) and get (tenant 2) are compared.

Reply:
I don't see the difference, if the lists do no have 1 item in them, something went wrong. and the get by id should always go 404. 

Are you thinking there might be some asymmetry somewhere. Like Tenant2 could read Tenant1's data, but not visa versa?

Don't want to write unnecessary tests. and if we do want to be more granular, it should not be in the BDD test suite,

Reply:
those calls, list/search and get are completely different stacks. so you should be comparing the same stack. you are not confirming tenant 2's list function cannot return the value. likewise, you are not confirming that tenant 1's get works. you could do list 1 by alias like you have, get 1 using the id from list 1. then list 2 should fail (either nothing found with that alias, or that item has a different id than from list 1), and get 2 should fail as you have above.

Reply:
There are two tests, one is that the list endpoints only return their own contacts. 

The other grabs an ID from another tenants (arrange), and queries the get_by_id endpoint with a tenant that does not own that object (act), and asserts it's a 404. 

i have to get the id somehow, i could save it locally from the setup, but this seemed easier to understand, and easier to write.

Reply:
a more complete test actually calls both tenants with the same calls. in the above, you do not. it's not a lot of work to add 1 contact per tenant. call list for both tenants, their results should have 1 and their ids should be different. now you have 2 contacts from 2 tenants. then tenant 1 calls get with their contact id and tenant 2's contact id (404), likewise tenant 2 calls with their contact id and tenant 1's contact id (404). i wouldn't list by alias, i'd assert those after the fact to make sure each tenant's resulting contact is the one they are supposed to have.